gdx/src/com/badlogic/gdx/math/Rectangle.java

```diff
@@ -303,6 +303,60 @@ public float getAspectRatio () {
 		return (height == 0) ? Float.NaN : width / height;
 	}
 
+	/** Returns the bottommost coordinate of the rectangle.
+	 * @return The bottommost coordinate of the rectangle. */
+	public float getBottom () {
+		return y;
+	}
+
+	/** Sets the Y coordinate so that the bottom of the rectangle is at the given value.
+	 * @param y The value for the bottom. */
+	public void setBottom (float y) {
+		this.y = y;
+	}
+
+	/** Returns the topmost coordinate of the rectangle.
+	 * @return The topmost coordinate of the rectangle. */
+	public float getTop () {
+		return y + height;
+	}
+
+	/** Sets the Y coordinate so that the top of the rectangle is at the given value.
+	 * @param y The value for the top. */
+	public void setTop (float y) {
+		this.y = y - height;
+	}
+
+	/** Returns the rightmost coordinate of the rectangle.
+	 * @return The rightmost coordinate of the rectangle. */
+	public float getRight () {
+		return x + width;
+	}
+
+	/** Sets the X coordinate so that the right of the rectangle is at the given value.
+	 * @param x The value for the right. */
+	public void setRight (float x) {
+		this.x = x - width;
+	}
+
+	/** Returns the leftmost coordinate of the rectangle.
+	 * @return The leftmost coordinate of the rectangle. */
+	public float getLeft () {
+		return x;
+	}
+
+	/** Sets the X coordinate so that the left of the rectangle is at the given value.
+	 * @param x The value for the left. */
+	public void setLeft (float x) {
+		this.x = x;
+	}
+
+	/** Calculates the center of the rectangle.
+	 * @return A vector with the center position. */
+	public Vector2 getCenter () {

```

Comment:
I think that I understand what you're saying. But still can't quite wrap my head around it. Let's say I have the following code in an EnemyShip class:

```
private void shoot(){
    Texture projectileSprite = Game1.GetProjectileSprite(ENEMY_SHIP_PROJECTILE_TYPE);
    Vector2 center = drawRectangle.getCenter();

    int projectileY = (int) (center.y + GameConstants.ENEMY_SHIP_PROJECTILE_OFFSET);
    Projectile projectile = new Projectile(ENEMY_SHIP_PROJECTILE_TYPE, projectileSprite, center.X, projectileY, -GetProjectileYVelocity());
    Game.AddProjectile(projectile);
}
```

This method will be called in every enemy ship that is currently in the game when a counter gets to a limit value (limit value that gets updated in an update method). 

This way the projectiles get shoot off the center of the enemy ships (and slightly down of its center).

I fail to see what improvement I could get creating a new Vector2 here and passing it to rectangle.getCenter(Vector2), instead of creating it inside rectangle.getCenter().

Again, I'm REALLY new to game programming so I may be doing this completely wrong. 

Maybe I should move this to another class and loop every ship using only one Vector2 for their centers?